backend/src/data/repositories/course-category-price/course-category-price.repository.ts

```diff
@@ -0,0 +1,36 @@
+import { CourseCategoryPriceGetAllItemResponseDto } from '~/common/types/types';
+import { CourseCategoryPrice as CourseCategoryPriceM } from '~/data/models/models';
+
+type Constructor = {
+  CourseCategoryPriceModel: typeof CourseCategoryPriceM;
+};
+
+class CourseCategoryPrice {
+  #CourseCategoryModel: typeof CourseCategoryPriceM;
+
+  public constructor({
+    CourseCategoryPriceModel: CourseCategoryModel,
+  }: Constructor) {
+    this.#CourseCategoryModel = CourseCategoryModel;
+  }
+
+  public getAll(): Promise<CourseCategoryPriceGetAllItemResponseDto[]> {
+    return this.#CourseCategoryModel
+      .query()
+      .withGraphJoined('category')
+      .castTo<CourseCategoryPriceGetAllItemResponseDto[]>()
+      .execute();
+  }
+
+  public getById(
+    id: number,
+  ): Promise<CourseCategoryPriceGetAllItemResponseDto | null> {
+    return this.#CourseCategoryModel
+      .query()
+      .findById(id)
+      .castTo<CourseCategoryPriceGetAllItemResponseDto>()
+      .execute();
+  }
```

Comment:
Does our request return null if it found nothing? Am I wrong?

Reply:
In the case you wrote, it'll return element or undefined
TS doesn't throw any errors, because you are casting to `CourseCategoryPriceGetAllItemResponseDto`, and TS thinks, that you always return `CourseCategoryPriceGetAllItemResponseDto`

Reply:
I think `findById` returns `undefined` if there is no result